Snowflake.Data/Client/SnowflakeDbDataReader.cs

```diff
@@ -255,6 +257,75 @@ public override int GetValues(object[] values)
             return count;
         }
 
+        public T GetObject<T>(int ordinal)
+            where T : class, new()
+        {
+            try
+            {
+                var rowType = resultSet.sfResultSetMetaData.rowTypes[ordinal];
+                var fields = rowType.fields;
+                if (fields == null || fields.Count == 0)

```

Comment:
Should we also check if `type == object`?

Reply:
added